sdk/core/azure-core/azure/core/rest/_helpers.py

```diff
@@ -104,8 +104,22 @@ def set_urlencoded_body(data, has_files):
     return default_headers, body
 
 
+def has_files_array(files):
+    return any(isinstance(f, list) for f in files.values())
+
+
+# when there are multi files in same field name, (str, (...)) instead of dict could avoid file lost
 def set_multipart_body(files):
```

Comment:
Why no type hints?

Reply:
About `set_multipart_body`? It has no type hints before this PR.